flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultSchedulerTest.java

```diff
@@ -968,6 +982,75 @@ public void testExceptionHistoryWithRestartableFailure() {
                 lessThanOrEqualTo(updateStateTriggeringJobFailureTimeframe.upperEndpoint()));
     }
 
+    @Test(expected = IllegalStateException.class)
+    public void testRestoringModifiedJobFromSavepointFails() throws Exception {
+        // create savepoint data
+        final long savepointId = 42L;
+        final OperatorID operatorID = new OperatorID();
+        final File savepointFile =
+                TestUtils.createSavepointWithOperatorState(
+                        TEMPORARY_FOLDER.newFile(), savepointId, operatorID);
+
+        // set savepoint settings which don't allow non restored state
+        final SavepointRestoreSettings savepointRestoreSettings =
+                SavepointRestoreSettings.forPath(savepointFile.getAbsolutePath(), false);
+
+        // create a new operator
```

Comment:
```suggestion
        // create a new operator
        // this test will fail in the end due to the previously created Savepoint having a state for a given OperatorID that does not match any operator of the newly created JobGraph
```
(please correct me if I'm wrong with the suggested comment above)
I would suggest adding some comment about the rationals of the test here to make a reader understand why we expect the test to fail.

Reply:
This would apply to the AdaptiveSchedulerTest accordingly.

Reply:
Good point. Will add it.